### PR TITLE
aws_ec2 inventory plugin: backport/2.5/40240

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Backporting #40240

Fallback to IAM role credentials

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
aws_ec2

##### ANSIBLE VERSION
```
2.5.2
```

### CODE REVIEW

File: changelogs/fragments/aws_ec2_inventory_fallback_to_instance_role_credentials.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - Fallback to IAM role temporary credentials if none are explicitly provided for the aws_ec2 inventory plugin
```
Comment (ryansb) [inline changelogs/fragments/aws_ec2_inventory_fallback_to_instance_role_credentials.yaml]: `instance role STS credentials` would be more accurate for the changelog

**Review by ryansb — CHANGES_REQUESTED** [review]
Changelog nitpick, LGTM otherwise.

**Review by ryansb — APPROVED** [review]

**Review by nitzmahone — APPROVED** [review]

**Comment by nitzmahone** [comment]
rebuild_merge

**Comment by rseuchter** [comment]
This (backport of) the fix stopped my headaches concerning the aws_ec2 inventory plugin in Ansible 2.5. Great work!

**Pull request merged**